src/core/lib/security/security_connector/security_connector.cc

```diff
@@ -120,17 +120,21 @@ const tsi_peer_property* tsi_peer_get_property_by_name(const tsi_peer* peer,
 
 void grpc_channel_security_connector_add_handshakers(
     grpc_channel_security_connector* connector,
+    grpc_pollset_set* interested_parties,
     grpc_handshake_manager* handshake_mgr) {
   if (connector != nullptr) {
     connector->add_handshakers(connector, handshake_mgr);
+    connector->base.interested_parties = interested_parties;
```

Comment:
Instead of setting this, just pass it to `connector->add_handshakers()` directly.

It should not be necessary to store a pointer to `interested_parties` in the security connector itself; the security connector should just be passing it through.

Reply:
Just to confirm, since it will be an API change, it requires all security connector implementations (i.e., both internal and external) to comply with it. Correct?

Reply:
Another reason for storing the pointer to `interested_parties` in the security connector is that, as discussed in (https://github.com/grpc/grpc/pull/16695#issuecomment-426395710), I plan to migrate psp to use the `interested_parties` passed in during handshaker construction time (instead of handshake time), which requires it to retrieve it from somewhere (e.g., security connector) and I believe security connector is a good place to store it. Since each security connector will always have its own corresponding `interested_parties`, I think it is not a bad idea to store it in security connector itself for the sake of future uses. How do you think?

Reply:
Regarding PSP, the current implementation passes `interested_parties` via `grpc_handshaker_args` that is set during `grpc_handshake_manager_do_handshake`. If we decide to pass it during the handshaker construction time, we will not pass it via `grpc_handshaker_args` any more, and all I can think of is to get it either from security connector or `loas2_client_handshaker`.

Reply:
If we are going to pass it to the handshaker at construction time, can't the handshaker store it internally?

Reply:
Yes, I think handshaker can store it, instead of security connector. I will update the PR.

Reply:
Regarding the patching CL, I did not find how the changes in #16695 could break internal code. Although there are some changes in handshaker_factory, all of its callers are in third_party/grpc, and they have already been updated.

Reply:
Looks like you're right.  I thought we had some internal implementations of handshaker factory, but it looks like I was remembering wrong.

Sorry for steering you wrong about that.

Reply:
Thanks for the confirmation. I will keep in mind to prepare for a patching CL for this PR.